clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/AssignmentChangeEvent.java

```diff
@@ -16,33 +16,31 @@
  */
 package org.apache.kafka.clients.consumer.internals.events;
 
-import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.TopicPartition;
 
-import java.util.Collections;
-import java.util.Map;
+import java.util.Collection;
 
 public class AssignmentChangeEvent extends ApplicationEvent {
```

Comment:
To be able to know when the assignment change takes effect, we should probably have to make this extend CompletableApplicationEvent (<Void> probably? we just need to know that it completed)